Wrap calls to eventSender.monitorWheelEvents() in a UIHelper function

smfr opened the pull request:
#### 117ebe14e9c273e6be15f47cd04a7867d7145e74
<pre>
Wrap calls to eventSender.monitorWheelEvents() in a UIHelper function
<a href="https://bugs.webkit.org/show_bug.cgi?id=250058">https://bugs.webkit.org/show_bug.cgi?id=250058</a>
rdar://103857097

Reviewed by Ryosuke Niwa.

`eventSender.monitorWheelEvents()` will need to become async, so in preparation (and also to fix
scrolling tests with UI-side compositing), hide calls to `eventSender.monitorWheelEvents()` inside a
UIHelper function, which also waits for a rendering update.

Do other minor cleanup, like using `await UIHelper.waitForScrollCompletion()` where possible.

* LayoutTests/compositing/fixed-with-main-thread-scrolling.html:
* LayoutTests/fast/events/wheel/platform-wheelevent-in-scrolling-div.html:
* LayoutTests/fast/events/wheel/wheel-events-become-non-cancelable.html:
* LayoutTests/fast/repaint/resources/fixed-move-after-keyboard-scroll-iframe.html:
* LayoutTests/fast/scrolling/arrow-key-scroll-in-rtl-document.html:
* LayoutTests/fast/scrolling/latching/iframe-latch-small-deltas.html:
* LayoutTests/fast/scrolling/latching/iframe_in_iframe.html:
* LayoutTests/fast/scrolling/latching/latched-scroll-in-passive-region.html:
* LayoutTests/fast/scrolling/latching/latched-scroll-into-nonfast-region.html:
* LayoutTests/fast/scrolling/latching/latching-and-wheel-events.html:
* LayoutTests/fast/scrolling/latching/latching-stuck-to-main-page.html:
* LayoutTests/fast/scrolling/latching/nested-cross-axis-latch-expiration.html:
* LayoutTests/fast/scrolling/latching/overflow-hidden-on-one-axis.html:
* LayoutTests/fast/scrolling/latching/overflow-in-iframe-latching.html:
* LayoutTests/fast/scrolling/latching/scroll-div-latched-div.html:
* LayoutTests/fast/scrolling/latching/scroll-div-latched-mainframe.html:
* LayoutTests/fast/scrolling/latching/scroll-div-no-latching.html:
* LayoutTests/fast/scrolling/latching/scroll-div-with-nested-nonscrollable-iframe.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-fragment.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-in-overflow.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-latched-iframe.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-latched-mainframe.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-webkit1-latching-bug.html:
* LayoutTests/fast/scrolling/latching/scroll-latched-nested-div.html:
* LayoutTests/fast/scrolling/latching/scroll-nested-iframe.html:
* LayoutTests/fast/scrolling/latching/scroll-select-bottom-test.html:
* LayoutTests/fast/scrolling/latching/scroll-select-latched-mainframe.html:
* LayoutTests/fast/scrolling/latching/scroll-select-latched-select.html:
* LayoutTests/fast/scrolling/latching/scroll-snap-latching.html:
* LayoutTests/fast/scrolling/mac/adjust-scroll-snap-during-gesture.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-element.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-iframe.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-unscrollable-element.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-unscrollable-iframe.html:
* LayoutTests/fast/scrolling/mac/horizontal-overflow-trapping-small-deltas.html:
* LayoutTests/fast/scrolling/mac/keyboard-scrolling-terminates.html:
* LayoutTests/fast/scrolling/mac/keyboard-scrolling-with-mouse-scroll.html:
* LayoutTests/fast/scrolling/mac/rubberband-axis-locking.html:
* LayoutTests/fast/scrolling/mac/scroll-container-horizontally.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-fixed-element.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-iframe.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-with-overflow-hidden-and-layout.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-with-overflow-hidden.html:
* LayoutTests/fast/scrolling/mac/vertical-scroll-in-horizontal-scroller.html:
* LayoutTests/fast/scrolling/overflow-scroll-past-max.html:
* LayoutTests/fast/scrolling/programmatic-scroll-to-zero-zero.html:
* LayoutTests/fast/scrolling/programmatic-smooth-scroll-after-focus.html:
* LayoutTests/fast/scrolling/resources/overscroll-behavior-support.js:
(async mouseWheelScrollAndWait):
* LayoutTests/fast/scrolling/rtl-point-in-iframe.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-element.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-iframe.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-unscrollable-element.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-unscrollable-iframe.html:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async startMonitoringWheelEvents):
(window.UIHelper.async mouseWheelScrollAt):
(window.UIHelper.async statelessMouseWheelScrollAt):
* LayoutTests/scrollingcoordinator/mac/latching/horizontal-overflow-back-swipe-overscroll-behavior.html:
* LayoutTests/scrollingcoordinator/mac/latching/horizontal-overflow-back-swipe.html:
* LayoutTests/scrollingcoordinator/mac/latching/horizontal-overflow-in-vertical-overflow.html:
* LayoutTests/scrollingcoordinator/mac/latching/scrolling-select-should-not-latch-mainframe.html:
* LayoutTests/scrollingcoordinator/mac/nested-sticky-with-nonsticking-sticky-parent.html:
* LayoutTests/tiled-drawing/scrolling/fast-scroll-select-latched-mainframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/resources/mainframe-scroll-snap-test.js:
(async doScrollTest):
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-async-iframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-iframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-2d-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-async-overflow-stateless.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-async-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-borders.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-hidden-scrollbars.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-overflow-stateless.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-padding.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-rotated.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-then-proximity.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-phase-change-relatching.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-proximity-mainframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-proximity-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-scrolling-jumps-to-top.html:
* LayoutTests/tiled-drawing/scrolling/stateless-scrolling-no-rubber-band.html:

Canonical link: <a href="https://commits.webkit.org/258483@main">https://commits.webkit.org/258483@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1287ebe2f95e85bbc971f543cefc0e662f7672ee

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102087 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11232 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35156 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111412 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171588 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12206 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2143 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94467 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109148 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107867 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9341 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92614 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37155 "Passed tests") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91219 "Found 1 new API test failure: TestWebKitAPI.WebKit.QuotaDelegateNavigateFragment (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24091 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78880 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4789 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25519 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4898 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1963 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10955 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45015 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5818 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6646 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on LayoutTests/fast/scrolling/latching/scroll-div-no-latching.html:
```diff
@@ -20,6 +20,7 @@
     
 }
 </style>
+<script src="../../../resources/ui-helper.js"></script>
```
This test never seems to be using UIHelper?

rniwa commented on LayoutTests/fast/scrolling/mac/async-overscroll-behavior-element.html:
```diff
@@ -2,6 +2,7 @@
 <script src="../../../resources/testharness.js"></script>
 <script src="../../../resources/testharnessreport.js"></script>
 <script src="../resources/overscroll-behavior-support.js"></script>
+<script src="../../../resources/ui-helper.js"></script>
```
Doesn't seem like this test is using UIHelper?

rniwa commented on LayoutTests/fast/scrolling/mac/async-overscroll-behavior-iframe.html:
```diff
@@ -2,6 +2,7 @@
 <script src="../../../resources/testharness.js"></script>
 <script src="../../../resources/testharnessreport.js"></script>
 <script src="../resources/overscroll-behavior-support.js"></script>
+<script src="../../../resources/ui-helper.js"></script>
```
Ditto.

rniwa commented on LayoutTests/fast/scrolling/mac/async-overscroll-behavior-unscrollable-element.html:
```diff
@@ -3,6 +3,7 @@
 <script src="../../../resources/testharness.js"></script>
 <script src="../../../resources/testharnessreport.js"></script>
 <script src="../resources/overscroll-behavior-support.js"></script>
+<script src="../../../resources/ui-helper.js"></script>
```
Ditto.

smfr replied:
`resources/overscroll-behavior-support.js` pulls in UIHelper stuff with all these.

rniwa commented on LayoutTests/fast/scrolling/mac/async-overscroll-behavior-unscrollable-iframe.html:
```diff
@@ -3,6 +3,7 @@
 <script src="../../../resources/testharness.js"></script>
 <script src="../../../resources/testharnessreport.js"></script>
 <script src="../resources/overscroll-behavior-support.js"></script>
+<script src="../../../resources/ui-helper.js"></script>
```
Ditto.

rniwa commented on LayoutTests/fast/scrolling/mac/keyboard-scrolling-terminates.html:
```diff
@@ -15,7 +15,7 @@
+                await UIHelper.startMonitoringWheelEvents();
                 eventSender.keyDown("downArrow");
 
                 setTimeout(function() {
```
Can we use UIHelper.waitForScrollCompletion here as well?

smfr replied:
Maybe; I was reluctant to make changes that might affect test behavior.

rniwa commented on LayoutTests/fast/scrolling/mac/horizontal-overflow-trapping-small-deltas.html:
```diff
@@ -46,7 +46,7 @@
+            await UIHelper.startMonitoringWheelEvents();
             eventSender.mouseMoveTo(100, 100);
             eventSender.mouseScrollByWithWheelAndMomentumPhases(0, 0, "began", "none");
             eventSender.mouseScrollByWithWheelAndMomentumPhases(1, -5, "changed", "none");
```
Can we use UIHelper.waitForScrollCompletion below?

smfr replied:
Yes

rniwa commented on LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-element.html:
```diff
@@ -2,6 +2,7 @@
 <script src="../../resources/testharness.js"></script>
 <script src="../../resources/testharnessreport.js"></script>
 <script src="resources/overscroll-behavior-support.js"></script>
+<script src="../../resources/ui-helper.js"></script>
```
This test doesn't seem to use UIHelper either?

rniwa commented on LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-iframe.html:
```diff
@@ -2,6 +2,7 @@
 <script src="../../resources/testharness.js"></script>
 <script src="../../resources/testharnessreport.js"></script>
 <script src="resources/overscroll-behavior-support.js"></script>
+<script src="../../resources/ui-helper.js"></script>
```
Ditto.

rniwa commented on LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-unscrollable-element.html:
```diff
@@ -2,6 +2,7 @@
 <script src="../../resources/testharness.js"></script>
 <script src="../../resources/testharnessreport.js"></script>
 <script src="resources/overscroll-behavior-support.js"></script>
+<script src="../../resources/ui-helper.js"></script>
```
Ditto.

rniwa commented on LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-unscrollable-iframe.html:
```diff
@@ -2,6 +2,7 @@
 <script src="../../resources/testharness.js"></script>
 <script src="../../resources/testharnessreport.js"></script>
 <script src="resources/overscroll-behavior-support.js"></script>
+<script src="../../resources/ui-helper.js"></script>
```
Ditto.

rniwa commented on LayoutTests/resources/ui-helper.js:
```diff
@@ -70,6 +70,12 @@ window.UIHelper = class UIHelper {
         await UIHelper.animationFrame();
     }
 
+    static async startMonitoringWheelEvents(args)
```
Did you mean to say `...args`? It's strange to call a single argument args.

smfr replied:
It's only ever a single dictionary argument, but `...args` would work too.

smfr committed:
Wrap calls to eventSender.monitorWheelEvents() in a UIHelper function
https://bugs.webkit.org/show_bug.cgi?id=250058
rdar://103857097

Reviewed by Ryosuke Niwa.

`eventSender.monitorWheelEvents()` will need to become async, so in preparation (and also to fix
scrolling tests with UI-side compositing), hide calls to `eventSender.monitorWheelEvents()` inside a
UIHelper function, which also waits for a rendering update.

Do other minor cleanup, like using `await UIHelper.waitForScrollCompletion()` where possible.

* LayoutTests/compositing/fixed-with-main-thread-scrolling.html:
* LayoutTests/fast/events/wheel/platform-wheelevent-in-scrolling-div.html:
* LayoutTests/fast/events/wheel/wheel-events-become-non-cancelable.html:
* LayoutTests/fast/repaint/resources/fixed-move-after-keyboard-scroll-iframe.html:
* LayoutTests/fast/scrolling/arrow-key-scroll-in-rtl-document.html:
* LayoutTests/fast/scrolling/latching/iframe-latch-small-deltas.html:
* LayoutTests/fast/scrolling/latching/iframe_in_iframe.html:
* LayoutTests/fast/scrolling/latching/latched-scroll-in-passive-region.html:
* LayoutTests/fast/scrolling/latching/latched-scroll-into-nonfast-region.html:
* LayoutTests/fast/scrolling/latching/latching-and-wheel-events.html:
* LayoutTests/fast/scrolling/latching/latching-stuck-to-main-page.html:
* LayoutTests/fast/scrolling/latching/nested-cross-axis-latch-expiration.html:
* LayoutTests/fast/scrolling/latching/overflow-hidden-on-one-axis.html:
* LayoutTests/fast/scrolling/latching/overflow-in-iframe-latching.html:
* LayoutTests/fast/scrolling/latching/scroll-div-latched-div.html:
* LayoutTests/fast/scrolling/latching/scroll-div-latched-mainframe.html:
* LayoutTests/fast/scrolling/latching/scroll-div-no-latching.html:
* LayoutTests/fast/scrolling/latching/scroll-div-with-nested-nonscrollable-iframe.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-fragment.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-in-overflow.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-latched-iframe.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-latched-mainframe.html:
* LayoutTests/fast/scrolling/latching/scroll-iframe-webkit1-latching-bug.html:
* LayoutTests/fast/scrolling/latching/scroll-latched-nested-div.html:
* LayoutTests/fast/scrolling/latching/scroll-nested-iframe.html:
* LayoutTests/fast/scrolling/latching/scroll-select-bottom-test.html:
* LayoutTests/fast/scrolling/latching/scroll-select-latched-mainframe.html:
* LayoutTests/fast/scrolling/latching/scroll-select-latched-select.html:
* LayoutTests/fast/scrolling/latching/scroll-snap-latching.html:
* LayoutTests/fast/scrolling/mac/adjust-scroll-snap-during-gesture.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-element.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-iframe.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-unscrollable-element.html:
* LayoutTests/fast/scrolling/mac/async-overscroll-behavior-unscrollable-iframe.html:
* LayoutTests/fast/scrolling/mac/horizontal-overflow-trapping-small-deltas.html:
* LayoutTests/fast/scrolling/mac/keyboard-scrolling-terminates.html:
* LayoutTests/fast/scrolling/mac/keyboard-scrolling-with-mouse-scroll.html:
* LayoutTests/fast/scrolling/mac/rubberband-axis-locking.html:
* LayoutTests/fast/scrolling/mac/scroll-container-horizontally.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-fixed-element.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-iframe.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-with-overflow-hidden-and-layout.html:
* LayoutTests/fast/scrolling/mac/smooth-scroll-with-overflow-hidden.html:
* LayoutTests/fast/scrolling/mac/vertical-scroll-in-horizontal-scroller.html:
* LayoutTests/fast/scrolling/overflow-scroll-past-max.html:
* LayoutTests/fast/scrolling/programmatic-scroll-to-zero-zero.html:
* LayoutTests/fast/scrolling/programmatic-smooth-scroll-after-focus.html:
* LayoutTests/fast/scrolling/resources/overscroll-behavior-support.js:
(async mouseWheelScrollAndWait):
* LayoutTests/fast/scrolling/rtl-point-in-iframe.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-element.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-iframe.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-unscrollable-element.html:
* LayoutTests/fast/scrolling/sync-scroll-overscroll-behavior-unscrollable-iframe.html:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async startMonitoringWheelEvents):
(window.UIHelper.async mouseWheelScrollAt):
(window.UIHelper.async statelessMouseWheelScrollAt):
* LayoutTests/scrollingcoordinator/mac/latching/horizontal-overflow-back-swipe-overscroll-behavior.html:
* LayoutTests/scrollingcoordinator/mac/latching/horizontal-overflow-back-swipe.html:
* LayoutTests/scrollingcoordinator/mac/latching/horizontal-overflow-in-vertical-overflow.html:
* LayoutTests/scrollingcoordinator/mac/latching/scrolling-select-should-not-latch-mainframe.html:
* LayoutTests/scrollingcoordinator/mac/nested-sticky-with-nonsticking-sticky-parent.html:
* LayoutTests/tiled-drawing/scrolling/fast-scroll-select-latched-mainframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/resources/mainframe-scroll-snap-test.js:
(async doScrollTest):
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-async-iframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-iframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-2d-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-async-overflow-stateless.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-async-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-borders.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-hidden-scrollbars.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-overflow-stateless.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-padding.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-rotated.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-then-proximity.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-phase-change-relatching.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-proximity-mainframe.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-proximity-overflow.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-scrolling-jumps-to-top.html:
* LayoutTests/tiled-drawing/scrolling/stateless-scrolling-no-rubber-band.html:

Canonical link: https://commits.webkit.org/258483@main
